### PR TITLE
Migrate reflectionEquals tests

### PR DESCRIPTION
**Pull request by smaye81**
This migrates the reflection-equals tests to the `test-generated` package and removes its dependence on fixtures.

### CODE REVIEW

**Comment by smaye81** [comment]
@timostamm lmk what you think about this coverage. The spec already covers a bit of the equals logic. I added a few checks on the generated code for enums, oneofs, and maps.

File: packages/test-generated/spec/reflection-equals.spec.ts [hunk 1]
```diff
@@ -76,34 +79,59 @@ describe('reflectionEquals()', function () {
...
+            fields: OneofScalarMemberMessage.fields.map(normalizeFieldInfo),
+            options: {}
+        };
+        const msg = {
+            result: {
+                oneofKind: "value",
+                value: 42
+            }
+        } as PartialMessage<OneofScalarMemberMessage>;
+        const mt = new MessageType<UnknownMessage>(mi.typeName, mi.fields, mi.options);
```
Comment (timostamm) [inline packages/test-generated/spec/reflection-equals.spec.ts:94]: The fixtures did not provide MesssageTypes, and we had to use mergePartial. But here, we do have generated message types, and reduce the test to just reflectionEquals():

```suggestion
        const make = (): OneofScalarMemberMessage => ({
            result: {
                oneofKind: "value",
                value: 42
            }
        });
        const eq = reflectionEquals(
            OneofScalarMemberMessage,
            make() as unknown as UnknownMessage,
            make() as unknown as UnknownMessage
        );
        expect(eq).toBeTrue();
```

Ok, we rely on the generated interface matching the generated field info, but that still seems simpler than the fixtures. WDYT?

File: packages/test-generated/spec/reflection-equals.spec.ts [hunk 2]
```diff
@@ -76,34 +79,59 @@ describe('reflectionEquals()', function () {
...
+            enumField: 1,
+            repeatedEnumField: [0, 1, 2],
+            aliasEnumField: 1,
+            prefixEnumField: 2,
+        } as PartialMessage<EnumFieldMessage>;
+
+        const mt = new MessageType<UnknownMessage>(mi.typeName, mi.fields, mi.options);
+        const message = reflectionCreate(mt);
+        reflectionMergePartial(mi, message, msg);
+        expect(true).toBeTrue();
```
Comment (timostamm) [inline packages/test-generated/spec/reflection-equals.spec.ts:117]: I hope true is always true 🙂

Comment (smaye81) [reply packages/test-generated/spec/reflection-equals.spec.ts]: Ack, my mistake.

**Comment by smaye81** [comment]
@timostamm updated. I also had to remove any messages setting `BigInt` because of how we generate different types for those values based on plugin options. Lmk if there's a better way and/or a way to test the BigInt fields.

**Pull request merged**